src/BlockTypes/ProductQuery.php

```diff
@@ -210,4 +274,69 @@ private function get_filter_by_price_query() {
 			),
 		);
 	}
+
+	/**
+	 * Return a query that filters products by attributes.
+	 *
+	 * @return array
+	 */
+	private function get_filter_by_attributes_query() {
+		$attributes_filter_query_args = $this->get_filter_by_attributes_query_vars();
+
+		$queries = array_reduce(
+			$attributes_filter_query_args,
+			function( $acc, $query_args ) {
+				$attribute_name       = $query_args['filter'];
+				$attribute_query_type = $query_args['query_type'];
+
+				$attribute_value = get_query_var( $attribute_name );
+				$attribute_query = get_query_var( $attribute_query_type );
+
+				if ( empty( $attribute_value ) ) {
+					return $acc;
+				}
+
+				// It is necessary explode the value because $attribute_value can be a string with multiple values (e.g. "red,blue").
+				$attribute_value = explode( ',', $attribute_value );
+
+				$acc[] = array(
+					'taxonomy' => str_replace( AttributeFilter::FILTER_QUERY_VAR_PREFIX, 'pa_', $attribute_name ),
+					'field'    => 'slug',
+					'terms'    => $attribute_value,
+					'operator' => 'and' === $attribute_query ? 'AND' : 'IN',
+				);
+
+				return $acc;
+			},
+			array()
+		);
+
+		if ( empty( $queries ) ) {
+			return array();
+		}
+
+		return array(
+			// Ignoring the warning of not using meta queries.
+			// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
+			'tax_query' => array(
+				'relation' => 'AND',
+				$queries,
+			),
+		);
+	}
+
+	/**
+	 * Intersect arrays when both are not empty, otherwise merge them.
```

Comment:
Just a little grammar fix
```suggestion
	 * Intersect arrays when neither of them are empty, otherwise merge them.
```